tests/ui/perfherder/dropdown_menu_items_test.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import {
+  render,
+  screen,
+  fireEvent,
+  waitForElement,
+  getByRole,
+  shallow,
+} from '@testing-library/react';
+import { DropdownMenu } from 'reactstrap';
+
+import { createDropdowns } from '../../../ui/perfherder/FilterControls';
+import DropdownMenuItems from '../../../ui/shared/DropdownMenuItems';
+
+// const dropdownOptions = [
+//   {
+//     selectedItem: 'try',
+//     updateData: () => {},
+//     pinned: ['try', 'autoland', 'bogus'],
+//     options: ['mozilla-inbound', 'autoland', 'mozilla-central', 'try'],
+//     namespace: 'Projects',
+//   },
+// ];
+
+const selectedItem = 'try';
+const pinned = ['try', 'autoland', 'bogus'];
+const options = ['mozilla-inbound', 'autoland', 'mozilla-central', 'try'];
+const namespace = 'Projects';
+
+const dropdownMenuItems = () =>
+  render(
+    <DropdownMenuItems
+      selectedItem={selectedItem}
+      updateData={() => {}}
+      options={options}
+      pinned={pinned}
+      namespace={namespace}
+    />,
+  );
+
+test('pinned items are shown first', async () => {
+  const { getByRole, getByText } = dropdownMenuItems();
+  // render(createDropdowns(dropdownOptions));
+
+  const component = render(
+    <DropdownMenuItems
+      selectedItem={selectedItem}
+      updateData={() => {}}
+      options={options}
```

Comment:
Since you're already passing the `options` here, you don't need to mock a fetch call. You would if you were doing this in the component that calls the `repository` API so this will be easier to test :)